mlir/tools/mlir-pet/islNodeBuilder.cc

```diff
@@ -29,19 +29,72 @@ static int createIntFromIslExpr(isl::ast_expr expression) {
 // TODO: See how we can get location information.
 // TODO: handle degenerate loop (see isl_ast_node_for_is_degenerate)
 // TODO: See how to handle more complex expression in the loop.
+
+bool isInt(isl::ast_expr expression) {
+  if (isl_ast_expr_get_type(expression.get()) == isl_ast_expr_int)
+    return true;
+  return false;
+}
+bool isId(isl::ast_expr expression) {
+  if (isl_ast_expr_get_type(expression.get()) == isl_ast_expr_id)
+    return true;
+  return false;
+}
 void IslNodeBuilder::createFor(isl::ast_node forNode) {
-  auto lowerBound = forNode.for_get_init();
   auto increment = forNode.for_get_inc();
   auto iterator = forNode.for_get_iterator();
   auto iteratorId = iterator.get_id().to_str();
   auto upperBound = getUpperBound(forNode);
+  auto lowerBound = forNode.for_get_init();
+  auto incrementAsInt = createIntFromIslExpr(increment);
+  AffineForOp loop;
+  // symbolic upper bound
+  if (isInt(lowerBound) && isId(upperBound)) {
+    // get id from node type
+    auto upperBound_id = isl_ast_expr_get_id(upperBound.get());
+    // access string name
+    auto upperBound_processed = isl_id_get_name(upperBound_id);
+
+    auto lowerBound_processed = createIntFromIslExpr(lowerBound);
+    loop = MLIRBuilder_.createLoop(lowerBound_processed, upperBound_processed,
+                                   incrementAsInt);
+    isl_id_free(upperBound_id);
+  }
 
-  auto lowerBoundAsInt = std::abs(createIntFromIslExpr(lowerBound));
```

Comment:
The std::abs(was introduced for handling loops going downward. Did you have any test failures?